src/lib/POWER_DETECT/devPDET.cpp

```diff
@@ -29,41 +34,39 @@ static int start()
 
 static int timeout()
 {
-    if (!busyTransmitting) return PDET_BUSY_PERIOD;
+    if (!busyTransmitting) return PDET_BUSY_PERIODMS;
 
-    float newPdet = analogReadMilliVolts(GPIO_PIN_PA_PDET);
+    pdet_storage_t newPdetScaled = PDET_MV_SCALE(analogReadMilliVolts(GPIO_PIN_PA_PDET));
 
-    if (!busyTransmitting) return PDET_BUSY_PERIOD; // Check transmission did not stop during Pdet measurement.
+    if (!busyTransmitting) return PDET_BUSY_PERIODMS; // Check transmission did not stop during Pdet measurement.
 
-    if (!Pdet || currentPowerdBm != POWERMGNT::getPowerIndBm())
+    uint8_t targetPowerDbm = POWERMGNT::getPowerIndBm();
+    if (PdetMvScaled == 0 || lastTargetPowerdBm != targetPowerDbm)
     {
-        Pdet = newPdet;
-        currentPowerdBm = POWERMGNT::getPowerIndBm();
+        PdetMvScaled = newPdetScaled;
+        lastTargetPowerdBm = targetPowerDbm;
     }
     else
     {
-        Pdet = Pdet * 0.9 + newPdet * 0.1; // IIR filter
+        PdetMvScaled = (PdetMvScaled * 9 + newPdetScaled) / 10; // IIR filter
     }
 
-    float dBm = SKY85321_PDET_SLOPE * Pdet + SKY85321_PDET_INTERCEPT;
+    pdet_storage_t dBmScaled = PDET_MV_DESCALE(PDET_DBM_SCALE(SKY85321_PDET_SLOPE) * PdetMvScaled) + PDET_DBM_SCALE(SKY85321_PDET_INTERCEPT);
```

Comment:
I dont think this should have the PDET_MV_DESCALE.  Everything is being compared as scaled later.

Reply:
> I dont think this should have the PDET_MV_DESCALE. Everything is being compared as scaled later.

Definitely should descaled(scaled * scaled) = scaled.

Reply:
Yeah it has to descale it because it is double scaled. The constants that it compares against later are only scaled once. For example the Hysteresis is 0.7dBm scaled x 1000 = 700. If the slope is scaled by 1000 and the mV is scaled by 10, that makes the dBm scaled = Actual mV * 10 * 1000 (10,000 x scale) so one needs to be backed out to bring it down to the dBm scale alone. The alternative is to scale everything else by both the dBm and mV scales but I thought that got confusing that it was scaled twice.